### ISSUE TITLE
Cannot find pp for building

### ISSUE THREAD

**Issue by madaalch** [anoduck/Biber-OpenBSD#1]
**Few issues with guide**
-At section "Acquire dependencies and source"
 `pod::PerlDoc` => `Pod::Perldoc`
 `Text::BibTex` => `Text::BibTeX`
 `bibtex` could not be resolved, couldn't figure out why
-Section "Install btparse"
 Missing hyphen in `tar -zxvf ...`
 `cp -R ...` not `cp -r ...`
 `Build.PL` not `Build.pl`

After a few hours I got to the last step, everything seemes to have worked but cannot find pp file designated in the build.sh file line 21.

In directory perl5/perlbrew/build/perl-5.38.2/perl-5.382/ are pp.c pp.h pp.o - could I just compile those into the appropriate directory and retry build.sh?

Thanks for your work, appreciate it a lot.

**Comment by anoduck** [comment]
> **Few issues with guide** 
> -At section "Acquire dependencies and source" 
> `pod::PerlDoc` => `Pod::Perldoc`
Good catch.

> `Text::BibTex` => `Text::BibTeX`
> `bibtex` could not be resolved, couldn't figure out why

I pondered about this one a bit myself, but for a completely different reason, which you have now resolved. This is referring to btparse.

> -Section "Install btparse" Missing hyphen in `tar -zxvf ...`

This is a bad habit. 

> `cp -R ...` not `cp -r ...`

You must be a BSD animal! I alias `cp` to `gcp`. 

> `Build.PL` not `Build.pl`

An inconvenient Perl-ism. Why can't perl be like everyone else and use lowercase?

> After a few hours I got to the last step, everything seemes to have worked but cannot find pp file designated in the [build.sh](http://build.sh) file line 21.

Ooops... That because some silly duck forgot to quack it as one of the dependencies. FML.

``` sh
cpm install -g pp
```

> 
> In directory perl5/perlbrew/build/perl-5.38.2/perl-5.382/ are pp.c pp.h pp.o - could I just compile those into the appropriate directory and retry [build.sh](http://build.sh)?

Just use the above command. `pp` is Perl Packer.
 

> Thanks for your work, appreciate it a lot.

Thanks for catching those, and I discovered I used `$Home` rather than `$HOME`.

**Comment by anoduck** [comment]
@madaalch I updated the build.sh file and made corrections to the README.md

**Comment by madaalch** [comment]
Still need to change in the following section
"# Then install the remainder of the dependencies needed for the build.
`cpm install -g bibtex Readonly::XS Pod::Simple Pod::Simple::TranscodeSmart \
Pod::Simple::TranscodeDumb pod::PerlDoc Text::BibTex Text::CSV`"

- `Text::BibTex` to `Text::BibTeX` (uppercase X)
- `pod:PerlDoc` to `Pod:Perldoc` (Uppercase P in Pod and lowecase d in Doc)
- first entry after -g switch `bibtex` does not resolve; couldn't find anything according to it in cpan.org, what were you intending here?

Also in your build.sh script a few of the linked libraries can't be found on my build. Below is a list of them. Indented portions indicate the libraries I have, plus any notes.

```
- /usr/lib/libutil.so.13.1
-   /usr/lib/libutil.so.16.0
-   /usr/lib/libutil.so.17.0
- /usr/local/lib/eopenssl31/libcrypto.so.15.1
-   /usr/lib/libcrypto.so.50.0
-   /usr/lib/libcrypto.so.50.2
-   /usr/lib/libcrypto.so.52.0
-     Install openssl?	"it will conflict if library dependencies use LibreSSL libraries."
-     You have libcrypt.so.2 in repo Biber-OpenBSD/local/ ; unrelated?
- /usr/lib/libssl.so.47.4
-   /usr/lib/libssl.so.53.0
-   /usr/lib/libssl.so.53.2
-   /usr/lib/libssl.so.55.0
- /usr/local/lib/libgdbm.so.7.0
-   NOT FOUND
-     Install gdbm?
- /usr/lib/libz.so.6.0
-   /usr/lib/libz.so.7.0
```

Would it be legal/safe to include these in the ./local directory of the repo then link them appropriately in the build.sh script?

(**Note:**
There are some comments at the head of "biber/dist/linux_x86_64/build.sh" regarding searching for libraries and potential problems.)

> Thanks for catching those, and I discovered I used $Home rather than $HOME

  I noticed this too, but didn't throw any problems for me, but $HOME is surely better 

> You must be a BSD animal!

Yeah, my journey from Windows to OpenBSD involved only the shortest Linux stage!

Thanks again, I've seen a few OpenBSD users talking about how it's almost a deal-breaker for them, you've done great work!

**Comment by anoduck** [comment]
@madaalch I am sorry to hear you are having difficulties with those libraries, but grateful for the opportunity to work through it. 

Are you following CURRENT via the snapshots? Or are you following STABLE/RELEASE?

> You have libcrypt.so.2 in repo Biber-OpenBSD/local/ ; unrelated?

Most assuredly, yes. Truth be told, I "snatched" this from libraries belonging to Iridium, after much deliberation. It is the closest library to what biber originally requires. 

> gdbm

GDBM is the Gnu Database Manager, and it is required.

> OpenSSL

Yes, install OpenSSL. No, it will not conflict, and Yes, I am surprised you haven't already. OpenSSL comes from the OpenBSD project

> lib`<$WHATEVER>`.so.`<$SOMENUMBER>`

_For all the remaining libraries where there is a difference in the version numbering:_

Replace with the most recent version on your system. My system shares the versions you listed the next version up. Why those particular older versions were chosen at the time of script modification has yet to be determined, probably an out of date `mlocate.db`. I will have to build biber again using the newer releases. 

If the above does not work, then chose the lowest release version and using gnu or the standard `ln -sf` the lowest release to the version used in the script.

**Comment by madaalch** [comment]
> Are you following CURRENT via the snapshots? Or are you following STABLE/RELEASE?
Stable.

I followed your suggestions and it worked! I just used the most current versions of the libraries as you recommended, and it seems to work perfectly.
I'm marking this closed for now, but I'm going to open other issues until you correct some errors in the README.md which I have brought up a few times but you still haven't resolved.

Thanks again!
